src/main/java/net/sf/jabref/gui/fieldeditors/FileListEditor.java

```diff
@@ -218,13 +218,18 @@ public void actionPerformed(ActionEvent actionEvent) {
         menu.add(moveToFileDir);
         moveToFileDir.addActionListener(new MoveFileAction(frame, entryEditor, this, true));
 
-        JMenuItem deleteFile = new JMenuItem(Localization.lang("Delete local file"));
+        JMenuItem deleteFile = new JMenuItem(Localization.lang("Permanently delete local file"));
         menu.add(deleteFile);
         deleteFile.addActionListener(e -> {
             int row = getSelectedRow();
+
             // no selection
-            if (row != -1) {
+            if (row == -1) {
+                return;
+            }
 
+            int userConfirm = JOptionPane.showConfirmDialog(null, Localization.lang("Are you sure you want to delete this file?"), Localization.lang("Warning"), JOptionPane.YES_NO_OPTION);
+            if (userConfirm == JOptionPane.YES_OPTION) {
```

Comment:
Please don't use simple YES/NO Buttons: 
 http://www.uxdesignedge.com/2010/06/are-you-sure-how-to-write-effective-confirmations/

Better add explicit named Buttons Delete file permanently and Cancel:
As I just did this yesterday for another case, here is a sample code:

```
String[] options = {Localization.lang("Move file"), Localization.lang("Cancel")}; 
int dialogResult = JOptionPane.showOptionDialog(frame, "Move file to file directory" + fileDir.get(),
              "Move",
                   JOptionPane.INFORMATION_MESSAGE, JOptionPane.YES_NO_CANCEL_OPTION, null, options, options[0]);
            if (dialogResult == JOptionPane.YES_OPTION) {
```